internal/action/main_test.go

```diff
@@ -363,3 +373,15 @@ func TestWorkspaceRunTriggers(t *testing.T) {
 
 	assert.Len(t, triggers.Items, 1)
 }
+
+func randomString(n int) string {
```

Comment:
Similar to https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest#RandString

Nothing in stdlib to do this more concisely but plenty of package options if you wanted:

https://pkg.go.dev/search?q=random+string

I guess as far as "canonical," if there's no length constraint and `-` are allowed, UUIDs are probably the ideal random string.

Reply:
I don't see any limit specified in the [API docs](https://www.terraform.io/cloud-docs/api-docs/workspaces#create-a-workspace), but the UI enforces a 90 character limit. Our longest name comes in well under at 59 (`action-test-32ba52aa-67a4-4f33-83dd-09d2ea694518-production`) so I think we should be good